raster/r.surf.fractal/main.c

```diff
@@ -28,6 +28,11 @@ int main(int argc, char *argv[])
     struct Option *rast_out;   /* Structure for output raster     */
     struct Option *frac_dim;   /* Fractal dimension of surface.   */
     struct Option *num_images; /* Number of images to produce.    */
+    struct Option *seed;
+    struct Flag *s_flag;
```

Comment:
Just wondering, is the s-flag needed given that seed is optional?

Or is it added for consistency with current practice in other modules? Possibly redundand? If `seed` option is given, seed should be used, no?

And looking at the logic in lines 82 to 95 it seems that setting the s-flag has the same effect as not setting the s-flag...

So, maybe the s-flag can be dropped?

Reply:
I looked back at when this was [introduced in r.mapcalc](https://lists.osgeo.org/pipermail/grass-dev/2014-July/070150.html) by Glynn:

> My main objection to automatic seeding is that people will inevitably produce non-repeatable results without even realising it.

That's why user needs to choose -s flag or seed option explicitly. So the focus there was reproducibility. We could argue though that convenience is priority an in that case we could decide to drop the -s flag and not introduce it here.

Reply:
I wasn't following the development when the seed option was discussed for r.mapcalc. From an exterior point of view, the general trend of all software I have used yet that uses random numbers is that when you can set the seed, it is usually optional and for users who want know they want it reproducible. I almost always say it as an addition, not a requirement for just basic uses.

So, @Shashankss1205's interpretation is just like mine, and his implementation where not setting the `-s` flag results in a seed being generated, but with a warning with explanations, is my expected behaviour. Plus it doesn't break compatibility as well, and shows what was done. Maybe an even clearer message that makes the user aware and learn why setting a seed is wanted could be even better. If it's a pattern that we have for handling of -s flags, shouldn't these messages be common across GRASS and not specific to each module?

Reply:
Looks like the consensus so far is that -s flag could be dropped. @Shashankss1205 could you please rewrite this without the -s flag?

Reply:
+1 for dropping the _s_ flag. Outside of this PR, but this will require making it optional wherever it is used.

Reply:
Ok, I'll rewrite and send the code, my main aim was to keep the consistency of the previous code changes and abiding by the same rules. Will be happy to make the necessary changes.

Thank you for all of your's valuable insights.

Reply:
Are there any suggestions related to the updated code, I think I have removed all occurences of -s_flag from my code, and also I wanted to **seek some guidance** on how to work further for getting into **GSOC'24 with this organization.**

Reply:
Great! Please introduce yourself on [grass-dev mailing list](https://lists.osgeo.org/mailman/listinfo/grass-dev) and let us know which [idea](https://grasswiki.osgeo.org/wiki/GRASS_GSoC_Ideas_2024) you would like to work on so that we can further discuss it. See the tips on the [idea](https://grasswiki.osgeo.org/wiki/GRASS_GSoC_Ideas_2024) page as well. You can also start drafting the application based on the template specified [here](https://wiki.osgeo.org/wiki/Google_Summer_of_Code_Recommendations_for_Students).